shared/database_mysql.py

```diff
@@ -55,6 +56,28 @@ def execute(self, sql, args=None):
         except MySQLdb.Error as e:
             raise DatabaseException('Failed to execute `{sql}` with `{args}` because of `{e}`'.format(sql=sql, args=args, e=e))
 
+    def execute_with_reconnect(self, sql, args):
+        result = None
+        # Attempt to excute the query and reconnect 3 times, then give up
+        for _ in range(3):
+            try:
+                p = perf.start()
+                self.cursor.execute(sql, args)
+                perf.check(p, 'slow_query', (sql, args), 'mysql')
+                result = self.cursor.fetchall()
```

Comment:
Just return the result here if it works.

Reply:
It works but codacy complains about a for with else and no break. And with the break, it complains about no reachable code, so I left it like this, with the result variable to comply with both rules.